error with dns challenge with rimuhosting

### Welcome

- [X] Yes, I'm using a binary release within 2 latest releases.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).

### What did you expect to see?

when trying to create an LE cert using dns challenge and rimuhosting, it fails 

### What did you see instead?

i see the following error:

```
root@snow:~# RIMUHOSTING_API_KEY="redacted" lego --accept-tos --email='jason@examplexyz.com.au' --domains='snow.examplexyz.net.au' --dns rimuhosting run
2023/03/30 16:07:40 [INFO] [snow.examplexyz.net.au] acme: Obtaining bundled SAN certificate
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/215163879877
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] acme: Could not find solver for: tls-alpn-01
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] acme: Could not find solver for: http-01
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] acme: use dns-01 solver
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] acme: Preparing to solve DNS-01
2023/03/30 16:07:42 [INFO] [snow.examplexyz.net.au] acme: Cleaning DNS-01 challenge
2023/03/30 16:07:42 [WARN] [snow.examplexyz.net.au] acme: cleaning up failed: rimuhosting: failed to delete record for snow.examplexyz.net.au: expected element type <error> but have <html> 
2023/03/30 16:07:43 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/215163879877
2023/03/30 16:07:43 Could not obtain certificates:
	error: one or more domains had a problem:
[snow.examplexyz.net.au] [snow.examplexyz.net.au] acme: error presenting token: rimuhosting: failed to find record(s) for snow.examplexyz.net.au: expected element type <error> but have <html>

```



### How do you use lego?

Binary

### Reproduction steps

build latest lego
execute the following command: 
```RIMUHOSTING_API_KEY="redacted" lego --accept-tos --email='jason@examplexyz.com.au' --domains='snow.examplexyz.net.au' --dns rimuhosting run```


### Version of lego

```console
root@snow:~/go/bin# ./lego --version
lego version dev linux/amd64

```
```


### Logs

<details>

```2023/03/30 16:07:40 [INFO] [snow.examplexyz.net.au] acme: Obtaining bundled SAN certificate
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] AuthURL: https://acme-v02.api.letsencrypt.org/acme/authz-v3/215163879877
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] acme: Could not find solver for: tls-alpn-01
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] acme: Could not find solver for: http-01
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] acme: use dns-01 solver
2023/03/30 16:07:41 [INFO] [snow.examplexyz.net.au] acme: Preparing to solve DNS-01
2023/03/30 16:07:42 [INFO] [snow.examplexyz.net.au] acme: Cleaning DNS-01 challenge
2023/03/30 16:07:42 [WARN] [snow.examplexyz.net.au] acme: cleaning up failed: rimuhosting: failed to delete record for snow.examplexyz.net.au: expected element type <error> but have <html> 
2023/03/30 16:07:43 [INFO] Deactivating auth: https://acme-v02.api.letsencrypt.org/acme/authz-v3/215163879877
2023/03/30 16:07:43 Could not obtain certificates:
	error: one or more domains had a problem:
[snow.examplexyz.net.au] [snow.examplexyz.net.au] acme: error presenting token: rimuhosting: failed to find record(s) for snow.examplexyz.net.au: expected element type <error> but have <html>

```

</details>

### Go environment (if applicable)

<details>

```console
$ go version && go env
root@snow:~/go/bin# go version && go env
go version go1.19.6 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOENV="/root/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/root/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/root/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/lib/go-1.19"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/lib/go-1.19/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.19.6"
GCCGO="gccgo"
GOAMD64="v1"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
GOWORK=""
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build224019726=/tmp/go-build -gno-record-gcc-switches"

```

</details>

I tested with the version distributed with debian testing version and a version I built using the build instructions. I'm not familiar with go so I'm unsure how to get go to show proper version. I got the same error whether built from source or from the package. 

Debian package reports 4.9.1-1 - but lego installed that way also reports `lego version dev linux/amd64`

I then tested by installing using the suggested `go install github.com/go-acme/lego/v4/cmd/lego@latest` but that is what led to the version also reporting as 

```
root:~/go/bin# ./lego -v
lego version dev linux/amd64
```

I tried to build from git as per  the instructions on https://go-acme.github.io/lego/installation/ but I ran into  errors also:
```
$ make
BIN_OUTPUT: dist/lego
rm -rf dist/ builds/ cover.out
go generate ./...
Documentation for 116 DNS providers has been generated.
cli_help.toml updated
golangci-lint run
providers/dns/cloudxns/cloudxns.go:89:39: unused-parameter: parameter 'token' seems to be unused, consider removing or renaming it as _ (revive)
func (d *DNSProvider) Present(domain, token, keyAuth string) error {
                                      ^
providers/dns/cloudxns/cloudxns.go:101:39: unused-parameter: parameter 'token' seems to be unused, consider removing or renaming it as _ (revive)
func (d *DNSProvider) CleanUp(domain, token, keyAuth string) error {
                                      ^
.....

```

--- a/providers/dns/internal/rimuhosting/client.go
+++ b/providers/dns/internal/rimuhosting/client.go
@@ -14,7 +14,7 @@ import (
 // Base URL for the RimuHosting DNS services.
 const (
 	DefaultZonomiBaseURL      = "https://zonomi.com/app/dns/dyndns.jsp"
-	DefaultRimuHostingBaseURL = "https://rimuhosting.com/app/dns/dyndns.jsp"
+	DefaultRimuHostingBaseURL = "https://rimuhosting.com/dns/dyndns.jsp"
 )
 
 // Action names.

